test/Microsoft.Azure.Devices.Edge.Test.Common/IotHub.cs

```diff
@@ -174,33 +160,31 @@ public Task<Device> CreateEdgeDeviceIdentityAsync(string deviceId, Authenticatio
             Func<EventData, bool> onEventReceived,
             CancellationToken token)
         {
-            EventHubClient client = this.EventHubClient;
-            int count = (await client.GetRuntimeInformationAsync()).PartitionCount;
+            var options = new EventHubConnectionOptions()
+            {
+                TransportType = EventHubsTransportType.AmqpWebSockets
+            };
+            this.proxy.ForEach(p => options.Proxy = p);
+
+            await using var client = new EventHubConsumerClient(
+                EventHubConsumerClient.DefaultConsumerGroupName,
+                this.eventHubEndpoint,
+                new EventHubConsumerClientOptions { ConnectionOptions = options });
+
+            int count = (await client.GetPartitionIdsAsync(token)).Length;
             string partition = EventHubPartitionKeyResolver.ResolveToPartition(deviceId, count);
             seekTime = seekTime.ToUniversalTime().Subtract(TimeSpan.FromMinutes(2)); // substract 2 minutes to account for client/server drift
             EventPosition position = EventPosition.FromEnqueuedTime(seekTime);
-            PartitionReceiver receiver = client.CreateReceiver("$Default", partition, position);
-
-            var result = new TaskCompletionSource<bool>();
-            using (token.Register(() => result.TrySetCanceled()))
+            await foreach (PartitionEvent partitionEvent in client.ReadEventsFromPartitionAsync(
+                partition,
+                position,
+                token))
             {
-                receiver.SetReceiveHandler(
-                    new PartitionReceiveHandler(
-                        data =>
-                        {
-                            bool done = onEventReceived(data);
-                            if (done)
-                            {
-                                result.TrySetResult(true);
-                            }
-
-                            return done;
-                        }));
-
-                await result.Task;
+                if (onEventReceived(partitionEvent.Data))
```

Comment:
Assuming this is the part in our tests that validates it is actually coming from the right device id.

Reply:
Correct